src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -93,30 +95,70 @@ public class RequestExecutor : IDisposable
 
         public event EventHandler<(long RaftCommandIndex, ClientConfiguration Configuration)> ClientConfigurationChanged;
 
-        protected RequestExecutor(string databaseName, string apiKey)
+        public event Action<string> SucceededRequest;
+        public event Action<string> FailedRequest;
```

Comment:
If you are exposing the failure, it would be good to also exposed _what_ failed.